src/lib.rs

```diff
@@ -1,10 +1,48 @@
+//! Epoch-based memory reclamation.
+//!
+//! # Pointers
+//!
+//! Concurrent collections are built using atomic pointers. This module provides [`Atomic`], which
+//! is just a shared atomic pointer to a heap-allocated object. Loading an [`Atomic`] yields a
+//! [`Ptr`], which is an epoch-protected pointer through which the loaded object can be safely read.
+//!
+//! # Pinning
+//!
+//! Before an [`Atomic`] can be loaded, the current mutator must be [`pin`]ned. By pinning a mutator
```

Comment:
@stjepang I just assumed it was already discussed in the epoch-gc RFC (https://github.com/crossbeam-rs/rfcs/blob/master/text/2017-05-23-epoch-gc.md), but it actually isn't.  Sorry for that.

Each "realm" takes care of the garbages produced in it, but nothing else.  We can allocate a realm for a performance-critical subset of a program so that the code of concern is not penalized with (possibly) a lot of garbages produced elsewhere of the program.  In other words, we can restrict the effect of a bad-behaving neighbor to its own realm only.

We currently have only one global realm, but in the future maybe it is desirable to allow users to create their own realm.  Of course this should go through an RFC process, and I don't mean that we should do it.  I just wanted to explain why I prefer the term "mutator" to "thread" and use a generic lifetime for `Scope`, and why I organized the files (especially `global.rs`) in this way.